mlir/include/mlir/Interfaces/TilingInterface.td

```diff
@@ -63,8 +63,8 @@ def TilingInterface : OpInterface<"TilingInterface"> {
           The method returns the operation that is the tiled
           implementation.
         }],
-        /*retType=*/"FailureOr<TilingResult>",
-        /*methodName=*/"getTiledImplementation",
+        /*retType=*/"FailureOr<::mlir::TilingResult>",
+        /*methodName=*/"getTiledImplementationAsProducer",
```

Comment:
Leaving this as `getTiledImplementation` is fine. The naming is clear enough to me and best to minimize downstream API breakages.